waiter/src/waiter/token.clj

```diff
@@ -525,6 +534,10 @@
                           (not current-owner?)
                           existing-editor
                           (authz/run-as? entitlement-manager authenticated-user existing-editor))]
+
+        ;; do any custom validation
+        (post-validator-fn request new-token-data existing-token-metadata)
```

Comment:
I think these lines should be replaced by an invocation to `post-validator-fn`: https://github.com/twosigma/waiter/blob/7d2bef587cbdec269d7c34903e86a77f88bb0b34/waiter/src/waiter/token.clj#L474-L476

I'm not sure we need to pass `existing-token-metadata` to the validation function. It is responsible for validating the new inputs passed in the post argument.

Also, instead of passing the raw request, ew should pass a map with certain keys, e.g. the user.